### PR TITLE
Implement search with postgresql

### PR DESCRIPTION
**Pull request by tung2744**
ref #3395 

I will open another pr to create a script to reindex user in a given app id

### CODE REVIEW

**Review by louischan-oursky — COMMENTED** [review]
In elasticsearch, There is a hit if I enter `louis` in the search query, given that I have an account with email `louischan@oursky.com`. But in postgresql impl, there is no result.

File: pkg/lib/deps/providers.go [hunk 1]
```diff
@@ -104,6 +105,13 @@ func (p *RootProvider) NewAppProvider(ctx context.Context, appCtx *config.AppCon
 		cfg.SecretConfig.LookupData(config.DatabaseCredentialsKey).(*config.DatabaseCredentials),
 		loggerFactory,
 	)
+	searchDatabase := searchdb.NewHandle(
+		ctx,
+		p.DatabasePool,
```
[the remark this answers is not in the document]
Comment (tung2744) [reply pkg/lib/deps/providers.go:114]: Yes it is possible. Should we use a separated pool for search db?

**Comment by tung2744** [comment]
@louischan-oursky Updated, thanks!

> In elasticsearch, There is a hit if I enter louis in the search query, given that I have an account with email louischan@oursky.com. But in postgresql impl, there is no result.


This was implemented using a prefix search in elasticsearch, I tried to replicate the behavior in d2d4434.

**Comment by tung2744** [comment]
Replaced by https://github.com/authgear/authgear-server/pull/4890

**Pull request closed without merging**